src/Microsoft.TemplateEngine.Edge/Settings/SettingsLoader.cs

```diff
@@ -295,6 +332,21 @@ private void EnsureLoaded()
             }
 
             _isLoaded = true;
+
+            //first run execution
+            if (!_paths.Exists(_paths.BaseDir) || !_paths.Exists(_paths.FirstRunCookie))
```

Comment:
discussion point: should it be in lock section?

Reply:
Probably whole EnsureLoaded method should be locked :)

Reply:
I revised locking here - keep TemplateCache locking to you :) - not sure if it needs to be separate or not.